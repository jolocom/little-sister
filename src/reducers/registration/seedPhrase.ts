import { AnyAction } from 'redux'

export const seedPhrase = (state = '', action: AnyAction): string => {
  switch (action.type) {
    case 'SEEDPHRASE_SET':
      return action.value
    case 'SEEDPHRASE_CLEAR':
      return ''
    default:
      return state
  }
}
