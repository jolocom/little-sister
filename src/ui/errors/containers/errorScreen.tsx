import React, { useEffect } from 'react'
import { BackHandler, StyleSheet, Text, View } from 'react-native'
import { JolocomButton, Wrapper } from '../../structure'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { centeredText, fontMain, fontMedium } from '../../../styles/typography'
import { Colors, Typefaces } from '../../../styles'
import { BlueCircleImage, RedCircleImage } from '../../../resources'
import { BP } from '../../../styles/breakpoints'
import { connect } from 'react-redux'
import { ThunkDispatch } from '../../../store'
import { navigationActions } from '../../../actions'
import { routeList } from '../../../routeList'

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: Colors.overflowBlack,
    justifyContent: 'space-between',
  },
  imageWrapper: {
    marginTop: '15%',
    width: '65%',
    height: '54%',
  },
  title: {
    ...Typefaces.title2,
    ...centeredText,
    fontFamily: fontMedium,
    color: Colors.sandLight090,
    paddingHorizontal: 10,
  },
  message: {
    ...centeredText,
    fontFamily: fontMain,
    lineHeight: BP({
      small: 20,
      medium: 20,
      large: 24,
    }),
    fontSize: BP({
      small: 18,
      medium: 18,
      large: 20,
    }),
    marginTop: 13,
    paddingHorizontal: 26,
    color: Colors.sandLight080,
  },
  interactButton: {
    marginTop: 14,
    borderColor: 'white',
    borderWidth: 0.5,
  },
  buttonSection: {
    flex: 0.3,
    width: '100%',
    paddingHorizontal: 20,
    justifyContent: 'center',
  },
  contentSection: {
    flex: 0.7,
    width: '100%',
    alignItems: 'center',
  },
})

export enum ImageType {
  Red = 'red',
  Blue = 'blue',
}

export interface ErrorScreenParams {
  title: string
  message: string
  image?: ImageType
  interact?: {
    label: string
    onInteract: () => void | Promise<void>
  }
  dismiss: {
    label: string
    onDismiss: () => void | Promise<void>
  }
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, ErrorScreenParams>
}

const ErrorScreenContainer: React.FC<Props> = props => {
  const {
    resetNavigation,
    navigation: {
      state: {
        params: { title, message, interact, dismiss, image },
      },
    },
  } = props

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', onBackButtonPress)

    return () => {
      BackHandler.removeEventListener('hardwareBackPress', onBackButtonPress)
    }
  }, [])

  const onBackButtonPress = (): boolean => {
    resetNavigation()
    return true
  }

  return (
    <Wrapper style={styles.wrapper}>
      <View style={styles.contentSection}>
        <View style={styles.imageWrapper}>
          {image === ImageType.Red ? <RedCircleImage /> : <BlueCircleImage />}
        </View>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.message}>{message}</Text>
      </View>
      <View style={styles.buttonSection}>
        <JolocomButton
          onPress={dismiss.onDismiss}
          transparent
          text={dismiss.label}
        />
        {interact && (
          <JolocomButton
            containerStyle={styles.interactButton}
            transparent
            onPress={interact.onInteract}
            text={interact.label}
          />
        )}
      </View>
    </Wrapper>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  resetNavigation: () =>
    dispatch(navigationActions.navigate({ routeName: routeList.AppInit })),
})

export const ErrorScreen = connect(
  null,
  mapDispatchToProps,
)(ErrorScreenContainer)
