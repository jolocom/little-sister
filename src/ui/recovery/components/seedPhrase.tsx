import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { Wrapper, JolocomButton } from 'src/ui/structure'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Spacing } from 'src/styles'

interface Props {
  seedPhrase: string
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: Colors.blackMain,
  },
  container: {
    paddingHorizontal: '5%',
  },
  noteSection: {
    marginTop: Spacing.LG,
    flex: 0.8,
    justifyContent: 'center',
  },
  note: {
    ...Typography.noteText,
    ...Typography.centeredText,
  },
  phraseSection: {
    flex: 1,
  },
  seedPhrase: {
    ...Typography.largeText,
    ...Typography.centeredText,
  },
  buttonSection: {
    marginTop: 'auto',
    marginBottom: 30,
  },
})

export const SeedPhrase: React.SFC<Props> = ({
  seedPhrase,
  handleButtonTap,
}: Props) => (
  <Wrapper style={styles.wrapper}>
    <View style={styles.container}>
      <View style={styles.noteSection}>
        <Text style={styles.note}>
          {I18n.t(
            strings.WRITE_THESE_WORDS_DOWN_ON_AN_ANALOG_AND_SECURE_PLACE,
          ) +
            '. ' +
            I18n.t(
              strings.WITHOUT_THESE_WORDS_YOU_CANNOT_ACCESS_YOUR_WALLET_AGAIN,
            ) +
            '.'}
        </Text>
      </View>
      <View style={styles.phraseSection}>
        <Text style={styles.seedPhrase}>{seedPhrase}</Text>
      </View>
      <View style={styles.buttonSection}>
        <JolocomButton
          onPress={handleButtonTap}
          text={I18n.t(strings.YES_I_WROTE_IT_DOWN)}
        />
      </View>
    </View>
  </Wrapper>
)
