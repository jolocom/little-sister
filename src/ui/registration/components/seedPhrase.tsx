import React from 'react'
import { StyleSheet } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { Container, Block, CenteredText } from 'src/ui/structure/'
import I18n from 'src/locales/i18n'

const { Button } = require('react-native-material-ui')

interface Props {
  seedPhrase: string
  checked: boolean
  handleButtonTap: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: JolocomTheme.primaryColorBlack,
  },
  phrase: {
    paddingLeft: '3%',
    paddingRight: '3%',
    color: JolocomTheme.primaryColorWhite,
    fontSize: 34,
    fontFamily: JolocomTheme.contentFontFamily,
    lineHeight: 38,
  },
  sideNote: {
    paddingLeft: '5%',
    paddingRight: '5%',
    lineHeight: 26,
    color: JolocomTheme.primaryColorSand,
    fontSize: JolocomTheme.labelFontSize,
    fontFamily: JolocomTheme.contentFontFamily,
  },
  footerButton: {
    position: 'absolute',
    bottom: '5%',
  },
  buttonContainer: {
    width: 225,
    height: 50,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontWeight: '100',
    fontSize: JolocomTheme.headerFontSize,
    color: JolocomTheme.primaryColorWhite,
    fontFamily: JolocomTheme.contentFontFamily,
  },
})

export const SeedPhrase: React.SFC<Props> = props => (
  <Container style={styles.container}>
    <Block flex={0.1}>
      <CenteredText
        style={styles.sideNote}
        msg={
          I18n.t('Write these words down on an analog and secure place') +
          '.\n ' +
          I18n.t('Without these words, you cannot access your wallet again') +
          '.'
        }
      />
    </Block>
    <Block flex={0.6}>
      <CenteredText style={styles.phrase} msg={props.seedPhrase} />
    </Block>
    <Block style={styles.footerButton}>
      <Button
        style={{ container: styles.buttonContainer, text: styles.buttonText }}
        onPress={props.handleButtonTap}
        raised
        upperCase={false}
        text={I18n.t('Yes, I wrote it down')}
      />
    </Block>
  </Container>
)
