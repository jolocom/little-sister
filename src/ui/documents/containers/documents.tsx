import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors } from 'src/styles'
import { Wrapper } from '../../structure'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

const styles = StyleSheet.create({
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
    paddingHorizontal: '5%',
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    if (isEmpty) {
      return (
        <Wrapper centered>
          <Text style={styles.emptyDocumentsText}>
            {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
          </Text>
        </Wrapper>
      )
    }

    return (
      <Wrapper>
        <Animated.View style={{ flex: 1 }}>
          {!otherIsEmpty && (
            <DocumentViewToggle
              showingValid={this.state.showingValid}
              onTouch={this.handleToggle}
            />
          )}
          <ScrollView
            // to scroll to top upon changing card
            ref={ref => (this.ScrollViewRef = ref)}
            contentContainerStyle={
              isEmpty ? { flex: 1 } : { paddingBottom: 110 }
            }
            scrollEnabled={!isEmpty}
          >
            {this.state.showingValid ? (
              <DocumentsCarousel
                documents={displayedDocs}
                activeIndex={this.state.activeDocumentIndex}
                onActiveIndexChange={this.onActiveIndexChange.bind(this)}
              />
            ) : (
              <DocumentsList
                documents={displayedDocs}
                onDocumentPress={openDocumentDetails}
              />
            )}
          </ScrollView>
        </Animated.View>
      </Wrapper>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
