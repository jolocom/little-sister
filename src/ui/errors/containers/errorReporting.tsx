import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { JolocomButton, Wrapper } from '../../structure'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import { SectionWrapper } from '../components/sectionWrapper'
import { NavigationSection } from '../components/navigationSection'

interface PaymentNavigationParams {
  error?: AppError | Error
  previousScreen: routeList
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState('')
  const [description, setDescription] = useState('')
  const [contact, setContact] = useState('')
  const [currentInput, setInput] = useState(Inputs.None)
  const [toggleState, setToggle] = useState(false)
  const [selectedEmoji, setEmoji] = useState(Emoji.Empty)

  // NOTE send previous screen as props, since not all
  // error reports will have error objects

  // const { previousScreen } = navigation.state.params
  // const isBackButton = previousScreen === routeList.Settings
  const isBackButton = false

  // NOTE error reports can show up without an error,
  // hence a navigation prop (navigateTo) would be better than in the error object
  const navigateBack = () => {
    const error = props.navigation?.state?.params?.error
    const previousScreen = props.navigation?.state.params?.previousScreen
    if (error instanceof AppError) {
      navigateToScreen(error.navigateTo)
    } else {
      navigateToScreen(previousScreen || routeList.AppInit)
    }
  }

  const onSubmitReport = () => {
    const error = props.navigation?.state?.params?.error

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
      sendPrivateData: toggleState,
    }

    if (navigation) {
      reportError({ ...userReport, error })
      navigateBack()
    }
  }

  return (
    <Wrapper secondaryDark>
      <NavigationSection
        onNavigation={navigateBack}
        isBackButton={isBackButton}
      />
      <ScrollView overScrollMode="never">
        <SectionWrapper
          title={I18n.t(strings.TELL_US_THE_PROBLEM)}
          style={{ marginTop: 14 }}>
          <ChooseIssueSection
            currentInput={currentInput}
            pickedIssue={pickedIssue}
            onIssuePick={setIssue}
            setInput={setInput}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.CAN_YOU_BE_MORE_SPECIFIC)}>
          <DescriptionSection
            currentInput={currentInput}
            setInput={setInput}
            setDescription={setDescription}
            toggleState={toggleState}
            setToggle={setToggle}
            description={description}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.WANT_TO_GET_IN_TOUCH)}>
          <ContactSection
            onContactInput={setContact}
            currentInput={currentInput}
            setInput={setInput}
            contactValue={contact}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.RATE_THE_ISSUE)}>
          <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        </SectionWrapper>
        <JolocomButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
            marginHorizontal: 20,
            height: 56,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Wrapper>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
