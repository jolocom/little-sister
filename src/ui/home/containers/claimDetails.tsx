import React from 'react'
import { ClaimDetailsComponent } from 'src/ui/home/components/claimDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { accountActions } from 'src/actions'
import { ClaimsState } from 'src/reducers/account'
import Immutable from 'immutable'

interface ConnectProps {
  claims: ClaimsState
  saveClaim: () => void
  handleClaimInput: (fieldValue: string, fieldName: string) => void
}

interface Props extends ConnectProps {}

interface State {}

export class ClaimDetailsContainer extends React.Component<Props, State> {
  render() {
    return (
      <ClaimDetailsComponent
        saveClaim={this.props.saveClaim}
        handleClaimInput={this.props.handleClaimInput}
        selectedClaim={this.props.claims.selected}
      />
    )
  }
}

const mapStateToProps = (state: RootState) => {
  const claims = Immutable.fromJS(state.account.claims)
  return {
    claims: claims.toJS(),
  }
}

const mapDispatchToProps = (dispatch: Function) => ({
  saveClaim: () => {
    dispatch(accountActions.saveClaim())
  },
  handleClaimInput: (fieldValue: string, fieldName: string) => {
    dispatch(accountActions.handleClaimInput(fieldValue, fieldName))
  },
})

export const ClaimDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClaimDetailsContainer)
