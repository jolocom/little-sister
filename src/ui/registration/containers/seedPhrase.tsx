import React from 'react'
import { connect } from 'react-redux'
import { SeedPhrase as SeedPhraseComponent } from 'src/ui/registration/components/seedPhrase'
import { finishRegistration } from 'src/actions/registration'

interface ConnectProps {}

interface Props extends ConnectProps {
  navigation: { state: { params: { mnemonic: string } } }
  finishRegistration: () => void
}

interface State {
  checked: boolean
}

export class SeedPhraseContainer extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      checked: false,
    }
  }

  render() {
    const { mnemonic } = this.props.navigation.state.params
    return (
      <SeedPhraseComponent
        seedPhrase={mnemonic}
        checked={this.state.checked}
        handleButtonTap={this.props.finishRegistration}
      />
    )
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = (dispatch: Function) => ({
  finishRegistration: () => dispatch(finishRegistration()),
})

export const SeedPhrase = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SeedPhraseContainer)
