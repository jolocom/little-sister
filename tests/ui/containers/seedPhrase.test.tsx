import React from 'react'
import { SeedPhraseContainer } from 'src/ui/registration/containers/seedPhrase'
import { shallow } from 'enzyme'

describe('seedPhrase container', () => {
  it('mounts correctly and matches snapshot', () => {
    const fetchSeedPhrase = jest.fn()
    const clearSeedPhrase = jest.fn()

    const props = {
      fetchSeedPhrase,
      clearSeedPhrase,
      navigation: {
        state: {
          params: {
            mnemonic: 'mock seedPhrase',
          },
        },
      },
    }

    const rendered = shallow(<SeedPhraseContainer {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
