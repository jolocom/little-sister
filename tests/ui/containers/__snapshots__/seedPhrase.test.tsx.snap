// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seedPhrase container mounts correctly and matches snapshot 1`] = `
<SeedPhrase
  handleButtonTap={[Function]}
  seedPhrase="mock seedPhrase"
/>
`;
